hypershift-operator/metrics.go

```diff
@@ -182,6 +205,33 @@ func (m *hypershiftMetrics) observeHostedClusters(hostedClusters *hyperv1.Hosted
 				}
 			}
 		}
+
+		guestCloudResourcesDeletionTime := clusterGuestCloudResourcesDeletionTime(&hc)
```

Comment:
Given that this function is run on a periodic basis, it's possible that the HostedCluster is gone by the time you try to catch the guest cloud resources condition. Would it make sense to also cache this when you detect that the condition exists?

Reply:
>Given that this function is run on a periodic basis, it's possible that the HostedCluster is gone by the time you try to catch the guest cloud resources condition.

It's technically possible, I think that's acceptable in practice.

>Would it make sense to also cache this when you detect that the condition exists?

mm Not following, can you elaborate?

Reply:
I was thinking that we could capture the state of the condition in a cache before the cluster goes away, but then you don't know when it turned true anyway, so please ignore me :)